File: app/src/main/java/com/jonasgerdes/stoppelmap/usecases/schedule/EventDayFragmentAdapter.java
package com.jonasgerdes.stoppelmap.usecases.schedule;

import android.content.Context;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;

import com.jonasgerdes.stoppelmap.R;
import com.jonasgerdes.stoppelmap.model.map.MapEntity;

/**
 * Created by Jonas on 04.08.2016.
 */
public class EventDayFragmentAdapter extends FragmentPagerAdapter {

    private final String[] mPageTitles;
    private MapEntity mEntity = null;
    private String mStartEventUuid = null;

    public EventDayFragmentAdapter(Context context, FragmentManager fm) {
        super(fm);
        mPageTitles = context.getResources().getStringArray(R.array.days);
    }

    @Override
    public Fragment getItem(int position) {
        return EventDayFragment.newInstance(position, mEntity, mStartEventUuid);
    }

    @Override
    public int getCount() {
        return 6;
    }

    @Override
    public CharSequence getPageTitle(int position) {
        return mPageTitles[position];
    }

    public void setEntity(MapEntity entity) {
        mEntity = entity;
    }

    public void setStartEventUuid(String startEventUuid) {
        mStartEventUuid = startEventUuid;
    }
}

File: app/src/main/java/com/jonasgerdes/stoppelmap/usecase/event/overview/view/EventDayFragmentAdapter.kt
package com.jonasgerdes.stoppelmap.usecase.event.overview.view

import android.support.v4.app.Fragment
import android.support.v4.app.FragmentManager
import android.support.v4.app.FragmentPagerAdapter
import com.jonasgerdes.stoppelmap.App
import com.jonasgerdes.stoppelmap.R
import com.jonasgerdes.stoppelmap.model.entity.map.MapEntity
import com.jonasgerdes.stoppelmap.usecase.event.overview.view.event_day.EventDayFragment
import com.jonasgerdes.stoppelmap.util.asset.StringResourceHelper
import javax.inject.Inject

/**
 * @author Jonas Gerdes <dev@jonasgerdes.com>
 * @since 10.08.17
 */
class EventDayFragmentAdapter(fm: FragmentManager?, val location: MapEntity? = null)
    : FragmentPagerAdapter(fm) {

    @Inject lateinit var stringRes: StringResourceHelper

    init {
        App.graph.inject(this)
    }

    override fun getItem(position: Int): Fragment {
        return EventDayFragment.createInstance(position, location)
    }

    override fun getCount(): Int {
        return 6
    }

    override fun getPageTitle(position: Int): CharSequence {
        return stringRes.getArray(R.array.days)[position]
    }
}